packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,191 @@
+import {
+    BaseSignerWalletAdapter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface SlopeWallet {
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+        };
+    }>;
+    disconnect(): Promise<{
+        msg: string;
+    }>;
+    signTransaction(message: string): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signature: string;
+        }
+    }>;
+    signAllTransactions(messages: string[]): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signatures: string[];
+        }
+    }>;
+}
+
+interface SlopeWindow extends Window {
+    Slope?: any;
+}
+
+declare const window: SlopeWindow;
+
+export interface SlopeWalletAdapterConfig {
+    pollInterval?: number;
+    pollCount?: number;
+}
+
+export class SlopeWalletAdapter extends BaseSignerWalletAdapter {
+    private _connecting: boolean;
+    private _wallet: SlopeWallet | null;
+    private _publicKey: PublicKey | null;
+
+    constructor(config: SlopeWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get ready(): boolean {
+        return !!window.Slope;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._publicKey;
+    }
+
+    get autoApprove(): boolean {
+        return false;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+
+            if (!window.Slope) throw new WalletNotFoundError();
+
+            const wallet = this._wallet || new window.Slope();
```

Comment:
I'm not sure how Slope is implemented, but just checking -- are there any possible complications from instantiating `new window.Slope()` multiple times?

The adapter cannot rely on there being only one instantiation of `window.slope`, because the adapter itself may be instantiated multiple times, and therefore `this._wallet` will not be defined for subsequent instances of it.

Reply:
Don’t worry, there are no complications, thank you for your checked.

Reply:
As long as it's not a problem to instantiate more than once, can we do this? The issue here is that `connect` should ideally start from a fresh state every time.
```suggestion
            const wallet = new window.Slope();
```